docs/src/user-guide/tutorials/auction.md

```diff
@@ -0,0 +1,351 @@
+
+```rust
+contract Auction {
+    pub fn __init__() {}
+}
+```
+
+To run the auction you will need several state variables, some of which can be initialized at the time the contract is instantiated.
+You will need to track the address of the beneficiary so you know who to pay out to. You will also need to keep track of the highest bidder, and the amount they have bid. You will also need to keep track of how much each specific address has sent into the contract, so you can refund them the right amount if they decide to withdraw. You will also need a flag that tracks whether or not the auction has ended. The following list of variables will suffice:
+
+```sh
+auction_end_time: u256
+beneficiary: address
+highest_bidder: address
+highest_bid: u256
+pending_returns: Map<address, u256>
+ended: bool
+```
+
+Notice that variables are named using camel case (lower case, underscore separated, `like_this`). 
+[Addresses](../../spec/type_system/types/address.md) have their own type in Fe - it represents 20 hex-encoded bytes as per the Ethereum specification.
+
+The variables that expect numbers are given the `u256` type. This is an unsigned integer of length 256 bits. There are other [choices for integers](../../spec/type_system/types/numeric.md) too, with both signed and unsigned integers between 8 and 256 bits in length.
+
+The `ended` variable will be used to check whether the auction is live or not. If it has finished `ended` will be set to `true`. There are only two possible states for this, so it makes sense to declare it as a `bool` - i.e. true/false.
+
+The `pending returns` variable is a mapping between N keys and N values, with user addresses as the keys and their bids as values. For this, a `Map` type is used. In Fe, you define the types for the key and value in the Map definition - in this case, it is `Map<address, u256>`. Keys can be any `numeric` type, `address`, `boolean` or `unit`.
+
+Now you should decide which of these variables will have values that are known at the time the contract is instantiated. It makes sense to set the `beneficiary` right away, so you can add that to the constructor arguments.
+
+The other thing to consider here is *how* the contract will keep track of time. On its own, the contract has no concept of time. However, the contract does have access to the current block timestamp which is measured in seconds since the Unix epoch (Jan 1st 1970). This can be used to measure the time elapsed in a smart contract. In this contract, you can use this concept to set a deadline on the auction. By passing a length of time in seconds to the constructor, you can then add that value to the current block timestamp and create a deadline for bidding to end. Therefore, you should add a `bidding_time` argument to the constructor. Its type can be `u256`.
+
+When you have implemented all this, your contract should look like this:
+
+```rust
```

Comment:
Is there a reason that this can not use the `fe`  formatting and hence benefit from CI checks? Looks pretty self contained to me.

Reply:
no - tbh i nuked them all in favour of ``rust because some were upsetting the linter, but now I see I can run `validate_doc_examples.py` locally to audit the snippets.